### ISSUE TITLE
Role with privilege escalation calls second role as dependency, task in second role uses /root as tmpdir

### ISSUE THREAD

**Issue by jladdjr** [ansible/ansible#20706]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
N/A

##### ANSIBLE VERSION
```
ansible 2.2.1.0 (detached HEAD 89fdca0929) last updated 2017/01/26 13:12:44 (GMT -400)
  lib/ansible/modules/core: (detached HEAD 59c9a6d3c2) last updated 2017/01/19 10:40:14 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD c501e006ea) last updated 2017/01/19 10:40:14 (GMT -400)
  config file = /Users/jladd/.ansible.cfg
  configured module search path = Default w/o overrides
~/git/tower-qa/reproducer
```

##### CONFIGURATION
```
[defaults]
timeout=15
show_custom_stats = True
#remote_tmp=/tmp

[ssh_connection]
control_path = %(directory)s/%%h-%%r
```

##### OS / ENVIRONMENT
N/A

##### SUMMARY

When a playbook invokes a role using privilege escalation, and that role references a second role as a dependency, tasks in the second role use /root as tmpdir. If those tasks attempt to use the tmpdir though, the task will fail with a permission denied error. It seems like the second role wants to use the system's temp directory, but doesn't actually have escalated privileges.

##### STEPS TO REPRODUCE

Create the following:

*inventory*
```
ec2-54-145-8-245.compute-1.amazonaws.com ansible_ssh_user=centos
```

*playbook.yml*
```
---
- name: 'Call first role'
  hosts: 'all'
  become: true
  roles:
    - { role: first_role }
```

*roles/first_role/tasks/main.yml*
```
---

- name: First role
  debug:
   msg: 'first role'
```

*roles/first_role/meta/main.yml*
```
dependencies:
    - { role: second_role }
```

*roles/second_role/tasks/main.yml*
```
---
- name: install authorized key
  authorized_key:
    state: 'present'
    user: '{{ ansible_ssh_user }}'
    key: 'https://github.com/jladdjr.keys'
```

Call:
`$ ansible-playbook -i inventory.log playbook.yml`

##### EXPECTED RESULTS
Expect playbook to successfully add a public key to the instance.

##### ACTUAL RESULTS
Task fails with `Authentication or permission failure`.

```
Jamess-MacBook-Pro-2% ansible-playbook -vvvv -i inventory.log playbook.yml
Using /Users/jladd/.ansible.cfg as config file
Loading callback plugin default of type stdout, v2.0 from /Users/jladd/ansible/stable-2.2/lib/ansible/plugins/callback/__init__.pyc

PLAYBOOK: playbook.yml *********************************************************
1 plays in playbook.yml

PLAY [Call first role] *********************************************************

TASK [setup] *******************************************************************
Using module file /Users/jladd/ansible/stable-2.2/lib/ansible/modules/core/system/setup.py
<ec2-54-145-8-245.compute-1.amazonaws.com> ESTABLISH SSH CONNECTION FOR USER: centos
<ec2-54-145-8-245.compute-1.amazonaws.com> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=15 -o ControlPath=/Users/jladd/.ansible/cp/%h-%r ec2-54-145-8-245.compute-1.amazonaws.com '/bin/sh -c '"'"'echo ~root && sleep 0'"'"''
<ec2-54-145-8-245.compute-1.amazonaws.com> ESTABLISH SSH CONNECTION FOR USER: centos
<ec2-54-145-8-245.compute-1.amazonaws.com> SSH: EXEC ssh -vvv -C -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=centos -o ConnectTimeout=15 -o ControlPath=/Users/jladd/.ansible/cp/%h-%r ec2-54-145-8-245.compute-1.amazonaws.com '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /root/.ansible/tmp/ansible-tmp-1485455412.1-214412818491654 `" && echo ansible-tmp-1485455412.1-214412818491654="` echo /root/.ansible/tmp/ansible-tmp-1485455412.1-214412818491654 `" ) && sleep 0'"'"''
fatal: [ec2-54-145-8-245.compute-1.amazonaws.com]: UNREACHABLE! => {
    "changed": false,
    "msg": "Authentication or permission failure. In some cases, you may have been able to authenticate and did not have permissions on the remote directory. Consider changing the remote temp path in ansible.cfg to a path rooted in \"/tmp\". Failed command was: ( umask 77 && mkdir -p \"` echo /root/.ansible/tmp/ansible-tmp-1485455412.1-214412818491654 `\" && echo ansible-tmp-1485455412.1-214412818491654=\"` echo /root/.ansible/tmp/ansible-tmp-1485455412.1-214412818491654 `\" ), exited with result 1",
    "unreachable": true
}
        to retry, use: --limit @/Users/jladd/git/tower-qa/reproducer/playbook.retry

PLAY RECAP *********************************************************************
ec2-54-145-8-245.compute-1.amazonaws.com : ok=0    changed=0    unreachable=1    failed=0
```

**Comment by jladdjr** [comment]
On stable-2.2, git bisect points to  https://github.com/ansible/ansible/commit/89fdca092929fc39dbe864f6edd634f17f7d37a0

On devel, git bisect points to https://github.com/ansible/ansible/commit/10fa2cd0efcc6bd6634f261da79e66c2d9be1dc0

**Comment by bcoca** [comment]
can you test this?
```
diff --git a/lib/ansible/plugins/action/__init__.py b/lib/ansible/plugins/action/__init__.py
index b3a0213..f945f13 100644
--- a/lib/ansible/plugins/action/__init__.py
+++ b/lib/ansible/plugins/action/__init__.py
@@ -224,7 +224,7 @@ class ActionBase(with_metaclass(ABCMeta, object)):
             tmpdir =  self._remote_expand_user(C.DEFAULT_REMOTE_TMP)

         cmd = self._connection._shell.mkdtemp(basefile, use_system_tmp, tmp_mode, tmpdir)
-        result = self._low_level_execute_command(cmd, sudoable=False)
+        result = self._low_level_execute_command(cmd, sudoable=True)

         # error handling on this seems a little aggressive?
         if result['rc'] != 0:
```

**Comment by jladdjr** [comment]
Still failing:
https://gist.github.com/jladdjr/aaa672799cf2e19d086950cef176f121

**Comment by bcoca** [comment]
fixed via 0a8b856102bfcc54b11027e1951c77bcb8ec3320

**Comment by MiLk** [comment]
#20332 is happening again.
